src/MSALWrapper/TaskExecutor.cs

```diff
@@ -31,7 +31,17 @@ internal static async Task<T> CompleteWithin<T>(ILogger logger, TimeSpan timeout
             source.CancelAfter(timeout);
             try
             {
-                return await getTask(source.Token).ConfigureAwait(false);
+                Task<T> mainTask = getTask(source.Token);
+                Task wrapperTask = Task.Run(() => mainTask);

```

Comment:
Can we test if this works without needing ` wrapperTask `

```
                Task wrapperTask = Task.Run(() => mainTask);
                await Task.WhenAny(Task.Delay(timeout), mainTask);
```

Reply:
I had to had `wrapperTask` because the `Task.WhenAny` takes an enumerable of `Task<T>` which != `Task`. 
`mainTask` is our `Task<T>` that we want to return after awaiting while the `wrapperTask` gives us a non-generic `Task` to race the delay against.

Reply:
Got it!